### PR TITLE
Fix #1422 Optimized Website

### LINKED ISSUE #1422: Test performance of EvalAI
**Actual Behaviour**

Currently the page is not optimized and is likely to deliver a slow user experience. 

**Expected Behaviour** 

Optimized Website.

**Would you like to work**

Yes.

@RishabhJain2018 

capture

### PR DESCRIPTION
**Pull request by gautamjajoo**
With reference to #1422 , done the changes in gulpfile.js ,compressed images and concatenated JS Controllers file into one.
@aka-jain

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -410,3 +411,33 @@ gulp.task('prod', function(callback) {
...
+    .pipe(gulp.dest('frontend/src/images'));
+});
+
+//Image Optimization of Organization Folder
+var imagemin = require('gulp-imagemin');
+var cache = require('gulp-cache');
+ gulp.task('imagesorgan', function() {
+  return gulp.src('frontend/src/images/organizations/**/*')
+    .pipe(cache(imagemin({ optimizationLevel: 5, progressive: true, interlaced: true })))
+    .pipe(gulp.dest('frontend/src/images/organizations'));
```
Comment (aka-jain) [inline gulpfile.js]: Src should be Dist folder.

File: gulpfile.js [hunk 2]
```diff
@@ -410,3 +411,33 @@ gulp.task('prod', function(callback) {
...
     runSequence('dev', 'connect', 'watch', 'test:watch', callback);
 });
+
+//Image Optimization
+var imagemin = require('gulp-imagemin');
+var cache = require('gulp-cache');
+ gulp.task('images', function() {
+  return gulp.src('frontend/src/images/**/*')
+    .pipe(cache(imagemin({ optimizationLevel: 5, progressive: true, interlaced: true })))
+    .pipe(gulp.dest('frontend/src/images'));
```
Comment (aka-jain) [inline gulpfile.js]: Minification should be done in dist folder.

File: gulpfile.js [hunk 3]
```diff
@@ -410,3 +411,33 @@ gulp.task('prod', function(callback) {
...
+    .pipe(gulp.dest('frontend/src/images/organizations'));
+});
+
+ // Concatenate & Minify JS
+gulp.task('scripts', function() {
+    return gulp.src('frontend/src/js/controllers/*.js')
+      .pipe(concat('controllers.js'))
+        .pipe(rename({suffix: '.min'}))
+        .pipe(uglify())
+        .pipe(gulp.dest('frontend/src/js/controllers'));
```
Comment (aka-jain) [inline gulpfile.js]: Dest should be dist folder

File: gulpfile.js [hunk 4]
```diff
@@ -410,3 +411,33 @@ gulp.task('prod', function(callback) {
...
+});
+// Default Task
+gulp.task('default', ['scripts', 'images', 'imagesorgan']);
+
```
Comment (aka-jain) [inline gulpfile.js]: Add minified files only to production task, please let me know if you need any help.

Comment (aka-jain) [reply gulpfile.js]: @gautamjajoo ^^

File: gulpfile.js (hunk 1 above)
Comment (gautamjajoo) [reply gulpfile.js]: A new Dist folder should be made?? Because I cannot see any folder for that.

Comment (aka-jain) [reply gulpfile.js]: Have you done with the setup of EvalAI frontend according to read me? Also, have you started gulp server?

Comment (gautamjajoo) [reply gulpfile.js]: Yeah! I have set up. and I have done according https://andy-carter.com/blog/a-beginners-guide-to-the-task-runner-gulp

Comment (aka-jain) [reply gulpfile.js]: So, if you done it correctly then it should have created a dist folder in your frontend directory. 
Please check or what tasks you are running to start a server?

Comment (gautamjajoo) [reply gulpfile.js]: I will try to setup from the beginning once more and will update you !!

**Comment by aka-jain** [comment]
Any update @gautamjajoo ?

**Comment by gautamjajoo** [comment]
I will update you in 2-3 days.

**Comment by gautamjajoo** [comment]
@aka-jain Removed the previous code as the minification was already done. Added similar code for the images of organization folder.

File: gulpfile.js [hunk 5]
```diff
@@ -165,10 +166,17 @@ gulp.task('html', function() {
 // for image compression
 gulp.task('images', function() {
     return gulp.src('frontend/src/images/**/*')
-        .pipe(gulp_if(flags.production, imagemin({ optimizationLevel: 3, progressive: true, interlaced: true })))
+        .pipe(gulp_if(flags.production, imagemin({ optimizationLevel: 5, progressive: true, interlaced: true })))
         .pipe(gulp.dest('frontend/dist/images'));
 });
 
+// for image of organization folder
+gulp.task('imagesorgan', function() {
```
Comment (aka-jain) [inline gulpfile.js]: No need to write this task, as ** will take nested folders too in `images` task.
Please remove this task. @gautamjajoo

**Comment by aka-jain** [comment]
@gautamjajoo remove all your tasks from gulp and just keep the value of image optimization to be `5`. Rest looks good as considered for images.

**Comment by gautamjajoo** [comment]
@aka-jain Done!!

File: frontend/src/js/controllers/controllers.min.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
...ntPage="",n.isNext="",n.isPrev="",n.team.error=!1,n.showPagination=!1,n.hostTeamId=null,n.challengeHostTeamId=null,n.isExistLoader=!1,n.loaderTitle="",n.loaderContainer=angular.element(".exist-team-card"),n.startLoader=a.startLoader,n.stopLoader=a.stopLoader,n.activateCollapsible=function(){angular.element(".collapsible").collapsible()};var l={};l.url="hosts/challenge_host_team/",l.method="GET",l.token=i,l.callback={onSuccess:function(a){var t=a.status,r=a.data;200==t&&(n.existTeam=r,0===n.existTeam.count?(n.showPagination=!1,n.paginationMsg="No team exists for now, start by creating a new team!"):(n.activateCollapsible(),n.showPagination=!0,n.paginationMsg=""),e.deleteData("emailError"),null===n.existTeam.next?n.isNext="disabled":n.isNext="",null===n.existTeam.previous?n.isPrev="disabled":n.isPrev="",null!==n.existTeam.next?n.currentPage=n.existTeam.next.split("page=")[1]-1:n.currentPage=1,n.load=function(e){if(n.isExistLoader=!0,n.loaderTitle="",n.loaderContainer=angular.element(".exist-team-card"),n.startLoader("Loading Teams"),null!==e){var a={Authorization:"Token "+i};s.get(e,{headers:a}).then(function(e){var a=e.data;n.existTeam=a,null===n.existTeam.next?(n.isNext="disabled",
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/controllers.min.js]: I think this file is wrongly placed in this folder. Please remove it.

**Comment by gautamjajoo** [comment]
@deshraj @aka-jain @RishabhJain2018 Please review.

**Comment by RishabhJain2018** [comment]
@aka-jain @spyshiv Please take a look here!

**Comment by spyshiv** [comment]
What are the changes in this PR. Only Images? @RishabhJain2018

**Comment by gautamjajoo** [comment]
@spyshiv Yeah!!

**Comment by spyshiv** [comment]
@gautamjajoo: There is no changes in images.

**Pull request closed without merging**